### ISSUE TITLE
[Information] How to get a list of inactive followers (following you, but not liking)

### ISSUE THREAD

**Issue by nikartz** [InstaPy/InstaPy#2057]
At first this may seem like a lot of work, however I am no expert in python and sadly can't program something more efficient. 

I just came up with a way to find some ghost followers using InstaPy and instaloader in a combination. With InstaPy I export all likes of a profile and I use instaloader to export a list of my followers. I am sure this could also be done with InstaPy, but I couldn't get it to work. After that I compare both lists with a script and get a final list of all profiles, that are following, but have never liked. I now can manually block them (I don't want that to happen automatically, because I think it should only hit fake accounts). At this point there is a lot of manual work included in the workflow, but maybe this helps develop a new feature in InstaPy or someone in interested. The [full guide](https://github.com/instaloader/instaloader/issues/120#issuecomment-390405666) I posted on the instaloader Issues section, because (although I know InstaPy for quite some time) originally I wanted to use instaloader (I found some help online on how to export a follower-list with instaloader).

Hope this helps a few people!

[If this 'information' is in an inappropriate place, please remove/change it]

**Comment by uluQulu** [comment]
Hi @nikartz,
To say, getting those lists is already used in InstaPy xD
_But it is not available for standalone use, having another issue targeting the same request, I think I would spend some time to do that job._

BTW that idea is amazing, I have never thought of how to get ghost users ;)
After implementing standalone grabber, writing the feature you've requested can be done easily and it worths to be done.

Super idea!

**Comment by aclowkey** [comment]
I guess we would have to parse the activity to figure out if users have liked/commented/ in the past in the past X days/weeks. Maybe some of the activity feeds are huge so it could take time.

@uluQulu Can you point me to where these lists are being used?

**Comment by sionking** [comment]
There are many application out there for guest users, but if someone don't Like your post it does not mean he don't like you.
I think a better solution is to just create this list manually and let the account owner to block those followers one-by-one, and leave the accounts that are "good" although not liking.

**Comment by uluQulu** [comment]
@aclowkey,
Yes that's what I have also thought. But for huge accaunts it'll provide inefficient flow.
But considering the not-so-huge accaunts, it is a neat feature ;)

The `onlyNotFollowMe=True` condition of `unfollow_users` feature gets those entire lists of your `followers` and `following` and compares the results to get users who do not follow you :)

_A few months back I wrote a tool in which to get the list of all followers, I used that same technic but into a new standalone function to be used easily anywhere. You can see it here https://github.com/timgrossmann/InstaPy/issues/1732#issuecomment-376577125 and that standalone grabber- `get_watchers` is **gist**ed [here](https://gist.github.com/uluQulu/51469ded8663a15cb462127978598059)_

_For example,
you can get your all followers by running that **gist** above like this,_
```erlang
all_of_followers = get_watchers(self.browser, self.username, self.logger, self.logfolder)
```

**Comment by uluQulu** [comment]
@sionking 
True, it shall take a bit complex logic in order to design that resilient feature for preserving efficiency whilst purging ghosts that's where it must be fully exposed to the user by self explaining parameters otherwise it is expected to be prone to pure confusion and misuse.

**Comment by dhabjan** [comment]
Hi,
if you intend to start some programming based on this topic, you might consider this three inputs:
- most of the "ghost" accounts, those that did no liking of your photos are just small accounts where IG AI didn't actually show your photo. What you need to do with them is let them know you exist to increase the chance of interaction. So I would prefer to find such and tell my script to do x number of likes on their last 10 posts.
- if the users havent posted anything in the last 3 months, they are probably inactive. It would be good to have some criteria based on when was the last post or how many post alltogether an account has (those with only 10 posts in 1 year are probably often inactive)
- most of the users that did not like my photos and who have over let's say 2000 account which they follow are bots. Even if they are not, the probably will not notice my posts since they have 2000+ accounts on their feed. Here I'd want to see the list and decide if I want to block them.

So, the logic would be:
Did the user like any of my last xz photos?
if not, does the user have more/less than  xy following (and/or xy followers as possibility?)
Did user post anything in last xy days?
If a, like the last xy photos of user (if there is any photo that can be liked)
If b, suggest blocking.

I hope you get what I meant :)

**Comment by uluQulu** [comment]
great overview @dhabjan 
It's a neat point I haven't thought of,
>most of the "ghost" accounts, those that did no liking of your photos are just small accounts where IG AI didn't actually show your photo

_And interestingly, those small accounts are the most genuine accaunts_ 😂

Maybe you could implement it, are you also a programmer? :P

**Comment by dhabjan** [comment]
sadly I'm an ex-unix power user and current application&IT PM, I just know how to define and troubleshoot a problem :D 
no programming skills beyond basic q-basic from 30 years ago.

**Comment by uluQulu** [comment]
that's great @dhabjan 
UNIX is also my fav OS 🎉
You've diagnosed the situation very well, now it will be very easy to implement it!

**Comment by dhabjan** [comment]
thanks :)
If you (or someone) start to develop it, I hope it would't be too much of a problem to expand on a feature for liking the latest posts from all of your own followers (not feed, so not from you are following).
I could imagine and define a bit more complicated logic, but don't want to push it :) The simple feature would be a good enough start - and it could be combined with others.
